consensus/core/src/dag_state.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    sync::Arc,
+};
+
+use crate::{
+    block::{BlockAPI as _, BlockRef, VerifiedBlock},
+    context::Context,
+    storage::Store,
+};
+
+/// DagState provides the API to write and read accepted blocks from the DAG.
+/// The underlying blocks can be cached in memory or stored on disk.
+#[allow(unused)]
+pub(crate) struct DagState {
+    context: Context,
+
+    // Cached refs are never removed for now.
+    cached_refs: Vec<BTreeSet<BlockRef>>,
+
+    // Blocks only need to be cached until they are committed.
```

Comment:
Maybe we can give some slack for a few rounds so we avoid touching disk when helping slower nodes catching up